Mathlib/AlgebraicTopology/SimplicialSet/HomotopyCat.lean

```diff
@@ -0,0 +1,469 @@
+  S.map (SimplexCategory.δ (n := 0) 0).op f
+
+/-- The hom-types of the refl quiver underlying a simplicial set `S` are subtypes of `S _[1]`.-/
+def SSet.OneTruncation.Hom {S : SSet} (X Y : OneTruncation S) :=
+  {p : S _[1] // src p = X ∧ tgt p = Y}
+
+/-- A simplicial set `S` has an underlying refl quiver `SSet.OneTruncation S`.-/
+instance (S : SSet) : ReflQuiver (SSet.OneTruncation S) where
+  Hom X Y := SSet.OneTruncation.Hom X Y
+  id X := by
+    refine ⟨S.map (SimplexCategory.σ (n := 0) 0).op X, ?_, ?_⟩ <;>
+    · change (S.map _ ≫ S.map _) X = X
+      rw [← map_comp, (_ : _ ≫ _ = 𝟙 _)]; simp
+      show ({..} : Opposite _) = _; congr; ext i
+      let 0 := i
+      rfl
+
+/-- The functor that carries a simplicial set to its underlying refl quiver.-/
+def SSet.oneTruncation : SSet.{u} ⥤ ReflQuiv.{u,u} where
+  obj S := ReflQuiv.of (OneTruncation S)
+  map {S T} F := {
+    obj := F.app (op [0])
+    map := fun f => by
+      refine ⟨F.app (op [1]) f.1, ?_, ?_⟩
+      · change (F.app _ ≫ _) _ = _
+        rw [← F.naturality]
+        exact congrArg (F.app _) f.2.1
+      · change (F.app _ ≫ _) _ = _
+        rw [← F.naturality]
+        exact congrArg (F.app _) f.2.2
+    map_id := fun X => by
+      change ({..} : Subtype _) = {..}
+      congr
+      change _ = (F.app _ ≫ _) _
+      rw [← F.naturality]
+      rfl
+  }
+  map_id X := by rfl
+  map_comp f g := by rfl
+
+section
+variable {C : Type u} [Category.{v} C]
+
+theorem opstuff.{w} (V : Cᵒᵖ ⥤ Type w) {X Y Z : C} {α : X ⟶ Y} {β : Y ⟶ Z} {γ : X ⟶ Z} {φ} :
+    α ≫ β = γ → V.map α.op (V.map β.op φ) = V.map γ.op φ := by
+  rintro rfl
+  change (V.map _ ≫ V.map _) _ = _
+  rw [← map_comp]; rfl
+
+
+/-- An arrow `f : X ⟶ Y` in the refl quiver of a nerve induces an arrow in the category `C`.-/
+def SSet.OneTruncation.ofNerve.map {X Y : OneTruncation (nerve C)}
+    (f : X ⟶ Y) : X.left ⟶ Y.left :=
+  eqToHom (congrArg (·.left) f.2.1.symm) ≫ f.1.hom ≫ eqToHom (congrArg (·.left) f.2.2)
+
+/-- The refl prefunctor from the refl quiver underlying a nerve to the refl quiver underlying a
+category.-/
+def SSet.OneTruncation.ofNerve.hom : OneTruncation (nerve C) ⥤rq C where
+  obj := (·.left)
+  map := OneTruncation.ofNerve.map
+  map_id := fun X : ComposableArrows _ 0 => by
+    simp only [SimplexCategory.len_mk, map, nerve_obj, eqToHom_refl, comp_id, id_comp,
+      ReflQuiver.id_eq_id]
+    exact ComposableArrows.map'_self _ 0
+
+/-- The refl prefunctor from the refl quiver underlying a category to the refl quiver underlying
+a nerve.-/
+def SSet.OneTruncation.ofNerve.inv : C ⥤rq OneTruncation (nerve C) where
+  obj := (.mk₀ ·)
+  map := fun f => by
+    refine ⟨.mk₁ f, ?_⟩
+    constructor <;> apply ComposableArrows.ext <;>
+      simp [SimplexCategory.len] <;> (exact fun 0 ↦ rfl)
+  map_id := fun X : C => Subtype.ext <| by
+    simp; apply ComposableArrows.ext <;> simp
+    · rintro _ rfl; simp; rfl
+    · intro; split <;> rfl
```

Comment:
If the `ext` attribute of suitable lemmas is activated, I think that these proofs could become completely automatic (`by aesop_cat`):
```suggestion
  map := fun f => by
    refine ⟨.mk₁ f, ?_⟩
    constructor <;> apply ComposableArrows.ext₀ <;> simp <;> rfl
  map_id := fun X : C => Subtype.ext <| by
    apply ComposableArrows.ext₁ <;> simp <;> rfl
```